includes/oauth/class-google-oauth.php

```diff
@@ -207,8 +213,14 @@ public static function api_google_auth_save_details( $request ) {
 		];
 		if ( isset( $request['refresh_token'] ) ) {
 			$auth_save_data['refresh_token'] = $request['refresh_token'];
+		} else {
+			Logger::log( 'Missing refresh token – credentials will expire after an hour an will have to be manually refreshed.' );
+		}
+		$auth_save_result = self::save_auth_credentials( $auth_save_data );
+		if ( is_wp_error( $auth_save_result ) ) {
+			return $auth_save_result;
 		}
-		if ( self::save_auth_credentials( $auth_save_data ) ) {
+		if ( $auth_save_result ) {
 			Logger::log( 'Credentials saved.' );
 			return \rest_ensure_response( [ 'status' => 'ok' ] );
 		} else {
```

Comment:
This check that returns `Could not save auth data for user.` is a result of `add_option()` not updating the database. The only reason for that to happen is if the option already exists, which will never be the case due to `self::remove_credentials()` always being called.

Reply:
`remove_credentials` might be removed from the method, and there might be a DB issue preventing the option from being saved. Returning the success status only if the option was saved is a more robust solution than assuming it succeeded.